### ISSUE TITLE
large files and wasm build

### ISSUE THREAD

**Issue by srothy** [buzz/mediainfo.js#25]
I am compiling 19.07 MediaInfoLib to wasm and using it. It works for small files, files smaller than my "chunk size". But if I have to iterate through chunks I get memory faults.

I'd like to try to use the functions I see in the gh-pages branch

mi.open_buffer_continue_goto_get_upper() 
and
mi.open_buffer_continue_goto_get_lower()

to see if this fixes the issue, as I had similar issues using the asm output that were solved by using these functions. but I cannot find them in the c++ source.

I am referring specifically to this commit:

https://github.com/buzz/mediainfo.js/commit/87dccc362a5136c83d76488f05235b7c90864391

Do they exist in c++, and if not, do you have any guidance for recreating them or for using the lib compiled to wasm?

thank you.

Soren Roth

**Comment by srothy** [comment]
I found the definition of upper and lower, however I am still having memory out of bounds issue.

**Comment by Akxe** [comment]
@srothy How did you solve this. And did you fork this to be wasm version?

**Comment by srothy** [comment]
I found somewhere that this is the definition:

` long mi_open_buffer_continue_goto_get_upper(){
    return (int64_t)mi.Open_Buffer_Continue_GoTo_Get() >> 32;
  }
  long mi_open_buffer_continue_goto_get_lower(){
    return (int64_t)mi.Open_Buffer_Continue_GoTo_Get();
  }  `

yes, I have forked to wasm using emscripten 1.38.30, and it works well. No patches were required. the main issue was that wasm has a fixed stackframe size of 5 MB, and so the 5 MB chunksize in the asm example was too big. the javascript to use it however I found to be a bit more challenging! I can provide more details if needed.
